include/dbrda_api.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2019 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef BINDINGS_DATA_ADAPTERS_DBRDA_API_H_
+#define BINDINGS_DATA_ADAPTERS_DBRDA_API_H_
+
+#include <errorcodes.h>
+#include <libdatabroker.h>
+
+#include <sys/uio.h>
+
+/**
+ * @typedef dbrDA_Handle_t
+ * @brief   Handle to the data adapter-specific structure
+ *
+ * Defines a general handle that the client lib can use to perform
+ * data adapter API calls and interaction.
+ */
+typedef void* dbrDA_Handle_t;
+
+
+typedef struct dbrDA_Request_chain
+{
+  struct dbrDA_Request_chain *_next;
+  DBR_Tuple_name_t _key;
+  int _sge_count;
+  struct iovec _value_sge[];   ///< variable size, needs to be last entry !!!
+} dbrDA_Request_chain_t;
+
+
+typedef struct dbrDA_api
+{
+  dbrDA_Handle_t (*initialize)(void);
+  int (*exit)( dbrDA_Handle_t );
+
+  /**
+   * pre-processing functions for read and write path
+   * they expect the request handle and will potentially modify the request directly
+   * if the alternative buffer is provided as non-NULL, then any modified data will
+   * be stored in that buffer
+   * pre-processing happens before handing the request over to the back-end
+   */
+  /**
+   * pre-write:
+   *  Allows to convert/modify the input tuple name (key) and SGE and
+   *  returns one key/value request or a chain for the backend to process
+   */
+  dbrDA_Request_chain_t* (*pre_write)( const DBR_Tuple_name_t, struct iovec*, const int );
+
+  /**
+   * pre-read:
+   *  Allows to manipulate the input key or initial data destination and
+   *  returns a key/value request or a chain for the backend to process
+   */
+  dbrDA_Request_chain_t* (*pre_read)( const DBR_Tuple_name_t, struct iovec*, const int );
+
+  /**
+   * post-processing functions for read and write path
+   * thex expect the request handle and will potentially modify the request directly
```

Comment:
typo.. not even worth mentioning (I feel ashamed for doing that!)

Reply:
I'll fix it. A catch is a catch. Nothing to be ashamed of. Thx.